oci/util.py

```diff
@@ -42,6 +44,27 @@ def urljoin(*parts):
     return '/'.join([first] + middle + [last])
 
 
+def sanitise_tag(tag: str) -> str:
+    '''
+    Additional build metadata as defined in SemVer can be added via `+` to the version. However,
```

Comment:
nit: considering this package is about OCI, it may a bit off to start talking about semver (from OCI's perspective, tags are just strings (with some limitations to allowed characters).

which makes it slightly problematic (imo) to add such logic to oci package :-/

maybe it would make more sense to pass-in a tag-processing callback (similar to git's clean/smudge callbacks)

Reply:
Changed it to callbacks and moved the respective sanitisation functions to the `cnudie.util` module instead.